pkg/action/install.go

```diff
@@ -152,22 +167,46 @@ func (i *Install) InstallAllSharedDeps(chrt *chart.Chart, settings *cli.EnvSetti
 		prefix = fmt.Sprintf("%*s", lvl*2, "- ")
 	}
 
-	for _, dep := range deps {
+	for _, dep := range sharedDeps {
+		switch i.OptionalDeps {
+		case OptionalDepsAll:
+			// install all deps, optional or not
+		case OptionalDepsNone:
+			if dep.IsOptional {
+				continue
+			}
+		case OptionalDepsAsk:
+			if dep.IsOptional {
+				reader := bufio.NewReader(os.Stdin)
+				question := eyecandy.ESPrintf(settings.NoEmojis, ":red_question_mark:Install optional shared dependency \"%s\" ?", dep.Name)
+				if !promptBool(question, reader, logger) {
+					continue
+				}
+			}
```

Comment:
If an optional dep is already installed, should a user be prompted about it or just told that the optional dep is already installed? The current flow doesn't seem right.

Reply:
indeed!
We fixed it to asking only when it isn't installed with https://github.com/rancher-sandbox/hypper/pull/157/commits/f5eb6db6adf5c2fe754f69add62d93e4ddb90380.